From d0987250dc321c9228522d78c761cb0460f443c5 Mon Sep 17 00:00:00 2001
From: Stefan Chulski <stefanc@marvell.com>
Date: Sun, 3 Jul 2016 11:22:34 +0300
Subject: [PATCH 0323/1345] fix: net: mvpp2x: fix in BM pool allocation

commit  e22aea7fc68358dcd760abe15f82be768cbb7a18 from
https://github.com/MarvellEmbeddedProcessors/linux-marvell.git

- allow to reach up to MVPP2_BM_POOLS_MAX_ALLOC_NUM
  inclusive

Change-Id: Id5dc933c576566b3b452e950f675acabf27bdb24
Signed-off-by: Stefan Chulski <stefanc@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/30896
Reviewed-by: Yuval Caduri <cyuval@marvell.com>
Tested-by: Star_Automation <star@marvell.com>
Reviewed-by: Hanna Hawa <hannah@marvell.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/net/ethernet/marvell/mvpp2x/mv_pp2x_main.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/net/ethernet/marvell/mvpp2x/mv_pp2x_main.c b/drivers/net/ethernet/marvell/mvpp2x/mv_pp2x_main.c
index efedb68..53bff79 100644
--- a/drivers/net/ethernet/marvell/mvpp2x/mv_pp2x_main.c
+++ b/drivers/net/ethernet/marvell/mvpp2x/mv_pp2x_main.c
@@ -408,7 +408,7 @@ int mv_pp2x_bm_pool_ext_add(struct device *dev, struct mv_pp2x *priv,
 	struct mv_pp2x_bm_pool *bm_pool;
 	struct mv_pp2x_hw *hw = &priv->hw;
 
-	if ((priv->num_pools + 1) >= MVPP2_BM_POOLS_MAX_ALLOC_NUM) {
+	if ((priv->num_pools + 1) > MVPP2_BM_POOLS_MAX_ALLOC_NUM) {
 		pr_err("Unable to add pool. Max BM pool alloc reached %d\n",
 		       priv->num_pools + 1);
 		return -ENOMEM;
-- 
1.7.9.5

